doc/source/notebooks/gpflow_2_upgrade/gpflow2_upgrade_guide.ipynb

```diff
@@ -0,0 +1,295 @@
+    "In GPFlow methods that used the `@autoflow` decorator, like for example `predict_f` and `predict_y` will previously have returned NumPy Arrays. These now return TensorFlow tensors. In many cases these can be used like NumPy arrays (they can be passed directly to many NumPy’s functions), but to actually turn them into numpy arrays you will need to call `.numpy()` on them.\n",
+    "\n",
+    "For example:\n",
+    "\n",
+    "<img src=\"numpy_new.png\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Parameter Values\n",
+    "\n",
+    "GPflow’s `Parameter.value` has changed from a property to a method.\n",
+    "\n",
+    "For example:\n",
+    "\n",
+    "<img src=\"param.png\">\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Model Class\n",
+    "\n",
+    "The `Model` class has been removed. A suitable replacement, for those models that do not wish to inherit from `GPModel`, may be `BayesianModel`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Periodic Base Kernel\n",
+    "\n",
+    "The base kernel for the `Periodic` kernel must now be specified explicitly. Previously the default was  `SquaredExponential`, so to maintain the same behaviour as before this must be passed-in to the `Periodic` kernel’s initialiser. \n",
+    "\n",
+    "For example:\n",
+    "\n",
+    "<img src=\"periodic_1.png\">\n",
+    "<img src=\"periodic_2.png\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Predict Full Covariance\n",
+    "\n",
+    "The `predict_f_full_cov` method has been removed from `GPModel`. Instead, pass `full_cov=True` to the `predict_f` method.\n",
+    "\n",
+    "For example:\n",
+    "\n",
+    "<img src=\"full_cov.png\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Data Types\n",
+    "\n",
+    "In some cases TensorFlow will try to figure out an appropriate data type for certain variables. If Python floats have been used TensorFlow may default these variables to `float32`, which can cause incompatibilities with GPflow, which defaults to using `float64`.\n",
+    "\n",
+    "To resolve this you can use `tf.constant` instead of a Python float, and explicitly specify the data type. For example:\n",
+    "\n",
+    "<img src=\"constant.png\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Float and Int Types\n",
+    "\n",
+    "In GPflow 2 `gpflow.settings.float_type` has changed to `gpflow.default_float()` and `gpflow.settings.int_type` has changed to `gpflow.default_int()`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Transforms\n",
+    "\n",
+    "These have been removed in favour of the tools in `tensorflow_probability.bijectors`. See for example [this Stackoverflow post](https://stackoverflow.com/q/58903446/5986907)."
```

Comment:
GPflow 2 still provides the `gpflow.utilities.triangular` alias for `tfp.bijectors.FillTriangular` and `gpflow.utilities.positive` which is configurable and can be either softplus or exp with an optional shift to ensure a lower bound that is larger than zero. Might be worth mentioning?

Reply:
I've added this